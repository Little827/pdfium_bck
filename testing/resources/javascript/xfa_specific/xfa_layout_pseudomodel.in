{{header}}
{{include ../../xfa_catalog_1_0.fragment}}
{{include ../../xfa_object_2_0.fragment}}
{{include ../../xfa_preamble_3_0.fragment}}
{{include ../../xfa_config_4_0.fragment}}
{{object 5 0}} <<
>>
stream
<template>
  <subform layout="tb" name="my_doc">
    <pageSet id="page" relation="orderedOccurrence">
      <occur initial="1" max="1" min="1"/>
      <pageArea id="Page1" name="Page1">
        <occur max="1" min="1"/>
        <contentArea h="100mm" w="200mm" x="0.25in" y="0.25in"/>
      </pageArea>
      <pageArea id="Page2" name="Page2">
        <occur max="1" min="1"/>
        <contentArea h="100mm" w="200mm" x="0.25in" y="0.25in"/>
      </pageArea>
    </pageSet>
    <event activity="docReady" ref="$host">
      <script contentType="application/x-javascript">
        function say(what) {
          xfa.host.messageBox(what, "xfa.host says");
        }
        function testReadProperty(prop) {
          try {
            var actual = xfa.layout[prop];
            say("property " + prop + " is " + actual);
          } catch (e) {
            say("unexpected error: " + e);
          }
        }
        function testROProperty(prop) {
          testReadProperty(prop);
          try {
            xfa.layout[prop] = "bogus";
            say("error: property " + prop + " was settable");
          } catch (e) {
            say("expected error setting " + prop + " is " + e);
          }
        }
        function testRWProperty(prop) {
          testReadProperty(prop);
          try {
            xfa.layout[prop] = "bogus";
            say("property " + prop + " was settable as expected");
          } catch (e) {
            say("error setting " + prop + " is " + e);
          }
        }
        testROProperty("ready");
        try {
         xfa.layout.absPage(1);
         xfa.layout.absPageCount();
         xfa.layout.absPageCountInBatch();
         xfa.layout.absPageInBatch(1);
         xfa.layout.absPageSpan(1);
         xfa.layout.h(100, 'pt');
         xfa.layout.page(1);
         xfa.layout.pageContent(1);
         xfa.layout.pageCount();
         xfa.layout.pageSpan(1);
         xfa.layout.relayout();
         xfa.layout.relayoutPageArea();
         xfa.layout.sheet(1);
         xfa.layout.sheetCount();
         xfa.layout.sheetCountInBatch();
         xfa.layout.sheetInBatch(1);
         xfa.layout.w(100, 'px');
         xfa.layout.x(100, 'mm');
         xfa.layout.y(100, 'bogus');
         say("finished testing methods");
        } catch (e) {
          say("error testing methods: " + e);
        }
      </script>
    </event>
    <subform layout="tb" name="subform_combox_0">
      <occur initial="1" max="10" min="0" name="occur_subform_combox_0">
      </occur>
      <field h="10mm" name="combox" w="40mm" x="10mm" y="10mm">
        <ui>
          <choiceList open="onEntry">
            <border>
              <edge/>
            </border>
          </choiceList>
        </ui>
        <items save="1">
          <text>apples</text>
          <text>bananas</text>
          <text>pears</text>
        </items>
        <value>
          <text>apples</text>
        </value>
      </field>
    </subform>
  </subform>
</template>
endstream
endobj
{{include ../../xfa_locale_6_0.fragment}}
{{include ../../xfa_postamble_7_0.fragment}}
{{xref}}
{{trailer}}
{{startxref}}
%%EOF
