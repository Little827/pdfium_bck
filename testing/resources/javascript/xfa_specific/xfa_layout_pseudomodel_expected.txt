xfa.host says: property ready is true
xfa.host says: expected error setting ready is Error: Unable to set ready value.
xfa.host says: finished testing methods
